File: BE/springboot/sotory/src/main/java/com/sotory/user/dto/response/UserInfoResponse.java
package com.sotory.user.dto.response;

import com.sotory.user.entity.User;

import java.util.UUID;

public record UserInfoResponse(
    UUID userId,
    String nickname,
    String birthday,
    String gender
) {
    public UserInfoResponse(User user){
        this(user.getUserId(), user.getNickname(), user.getBirthday(), String.valueOf(user.getGender()));
    }
}

File: sotory_aos/app/src/main/java/com/ssafy/sotory/data/dto/userinfo/UserInfoResponse.kt
package com.ssafy.sotory.data.dto.userinfo

import com.ssafy.sotory.domain.userinfo.Gender
import kotlinx.serialization.Serializable

@Serializable
data class UserInfoResponse(
    val userId: String,
    val nickname: String,
    val birthday: String?,
    val gender: Gender?
)